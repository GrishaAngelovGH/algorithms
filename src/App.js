import {
  HashRouter as Router,
  Routes,
  Route,
  Navigate
} from 'react-router-dom'

import Layout from './components/Layout'
import Sidebar from './components/Sidebar'

import LinearSearch from './components/algorithms/LinearSearch'
import BinarySearch from './components/algorithms/BinarySearch'
import BubbleSort from './components/algorithms/BubbleSort'
import QuickSort from './components/algorithms/QuickSort'
import BinaryTree from './components/algorithms/BinaryTree'
import CaesarCipher from './components/algorithms/CaesarCipher'
import FloodFill from './components/algorithms/FloodFill'

function App() {

  const linearSearchRoute = (
    <Layout sidebar={<Sidebar />}>
      <LinearSearch />
    </Layout>
  )

  const binarySearchRoute = (
    <Layout sidebar={<Sidebar />}>
      <BinarySearch />
    </Layout>
  )

  const bubbleSortRoute = (
    <Layout sidebar={<Sidebar />}>
      <BubbleSort />
    </Layout>
  )

  const quickSortRoute = (
    <Layout sidebar={<Sidebar />}>
      <QuickSort />
    </Layout>
  )

  const binaryTreeRoute = (
    <Layout sidebar={<Sidebar />}>
      <BinaryTree />
    </Layout>
  )

  const caesarCipherRoute = (
    <Layout sidebar={<Sidebar />}>
      <CaesarCipher />
    </Layout>
  )

  const floodFillRoute = (
    <Layout sidebar={<Sidebar />}>
      <FloodFill />
    </Layout>
  )

  return (
    <Router>
      <Routes>
        <Route exact path='/' element={<Navigate replace to='/linear-search' />} />
        <Route path='/linear-search' element={linearSearchRoute} />
        <Route path='/binary-search' element={binarySearchRoute} />
        <Route path='/bubble-sort' element={bubbleSortRoute} />
        <Route path='/quick-sort' element={quickSortRoute} />
        <Route path='/binary-tree' element={binaryTreeRoute} />
        <Route path='/caesar-cipher' element={caesarCipherRoute} />
        <Route path='/flood-fill' element={floodFillRoute} />
      </Routes>
    </Router>
  )
}

export default App