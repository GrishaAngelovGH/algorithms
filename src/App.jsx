import {
  HashRouter as Router,
  Routes,
  Route,
  Navigate
} from 'react-router-dom'

import Layout from './components/Layout'
import Sidebar from './components/Sidebar'

import LinearSearch from './components/algorithms/LinearSearch'
import BinarySearch from './components/algorithms/BinarySearch'
import BubbleSort from './components/algorithms/BubbleSort'
import QuickSort from './components/algorithms/QuickSort'
import BinaryTree from './components/algorithms/BinaryTree'
import CaesarCipher from './components/algorithms/CaesarCipher'
import FloodFill from './components/algorithms/FloodFill'

const layoutWith = Component => (
  <Layout sidebar={<Sidebar />}>
    <Component />
  </Layout>
)

function App() {
  return (
    <Router>
      <Routes>
        <Route exact path='/' element={<Navigate replace to='/linear-search' />} />
        <Route path='/linear-search' element={layoutWith(LinearSearch)} />
        <Route path='/binary-search' element={layoutWith(BinarySearch)} />
        <Route path='/bubble-sort' element={layoutWith(BubbleSort)} />
        <Route path='/quick-sort' element={layoutWith(QuickSort)} />
        <Route path='/binary-tree' element={layoutWith(BinaryTree)} />
        <Route path='/caesar-cipher' element={layoutWith(CaesarCipher)} />
        <Route path='/flood-fill' element={layoutWith(FloodFill)} />
      </Routes>
    </Router>
  )
}

export default App