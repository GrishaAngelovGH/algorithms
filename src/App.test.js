import { render, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

import App from './App'

test('should render Linear Search Algorithm', () => {
  render(<App />)

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Linear Search Algorithm$/)
})

test('should render Binary Search Algorithm', () => {
  render(<App />)

  act(() => {
    const binarySearchLink = document.querySelector("a[href='#/binary-search']")
    binarySearchLink.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Binary Search Algorithm$/)
})

test('should render Bubble Sort Algorithm', () => {
  render(<App />);

  act(() => {
    const bubbleSortLink = document.querySelector("a[href='#/bubble-sort']")
    bubbleSortLink.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Bubble Sort$/)
})

test('should render Quick Sort Algorithm', () => {
  render(<App />)

  act(() => {
    const bubbleSortLink = document.querySelector("a[href='#/quick-sort']")
    bubbleSortLink.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Quick Sort$/)
})

test.skip('should render Binary Tree Algorithms', () => {
  render(<App />);

  act(() => {
    const binaryTreeLink = document.querySelector("a[href='#/binary-tree']")
    binaryTreeLink.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Binary Tree Traversal Algorithms$/)
})

test('should render Caesar Cipher Algorithm', () => {
  render(<App />)

  act(() => {
    const bubbleSortLink = document.querySelector("a[href='#/caesar-cipher']")
    bubbleSortLink.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Caesar Cipher Algorithm$/)
})

test('should render Flood Fill Algorithm', () => {
  render(<App />)

  act(() => {
    const bubbleSortLink = document.querySelector("a[href='#/flood-fill']")
    bubbleSortLink.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  expect(screen.getAllByRole('heading')[1]).toHaveTextContent(/^Flood Fill Algorithm$/)
})
