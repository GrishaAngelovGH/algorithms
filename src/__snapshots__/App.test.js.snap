// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0 p-3"
      >
        <div
          class="col-md-3 col-lg-2"
        >
          <aside
            class="ps-sidebar-root css-1qqt2yo"
            data-testid="ps-sidebar-root-test-id"
            style="z-index: 100;"
            width="230px"
          >
            <div
              class="ps-sidebar-container css-1o85esu"
              data-testid="ps-sidebar-container-test-id"
            >
              <h3
                class="text-white bg-primary text-center p-2"
              >
                Algorithms
              </h3>
              <nav
                class="ps-menu-root css-vj11vy"
              >
                <ul
                  class="css-ewdv3l"
                >
                  <li
                    class="ps-menuitem-root ps-submenu-root ps-open css-1642kfo"
                  >
                    <a
                      class="ps-menu-button css-1bdadyq"
                      data-testid="ps-menu-button-test-id"
                    >
                      <span
                        class="ps-menu-label css-12w9als"
                      >
                        Searching
                      </span>
                      <span
                        class="ps-submenu-expand-icon css-1cuxlhl"
                      >
                        <span
                          class="css-jn69v9"
                          open=""
                        />
                      </span>
                    </a>
                    <div
                      class="ps-submenu-content css-edu7i6"
                      data-testid="ps-submenu-content-test-id"
                      open=""
                    >
                      <ul
                        class="css-ewdv3l"
                      >
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/linear-search"
                          level="1"
                        >
                          Linear Search
                        </a>
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/binary-search"
                          level="1"
                        >
                          Binary Search
                        </a>
                      </ul>
                    </div>
                  </li>
                  <li
                    class="ps-menuitem-root ps-submenu-root css-1642kfo"
                  >
                    <a
                      class="ps-menu-button css-1bdadyq"
                      data-testid="ps-menu-button-test-id"
                    >
                      <span
                        class="ps-menu-label css-12w9als"
                      >
                        Sorting
                      </span>
                      <span
                        class="ps-submenu-expand-icon css-1cuxlhl"
                      >
                        <span
                          class="css-honxw6"
                        />
                      </span>
                    </a>
                    <div
                      class="ps-submenu-content css-z5rm24"
                      data-testid="ps-submenu-content-test-id"
                    >
                      <ul
                        class="css-ewdv3l"
                      >
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/bubble-sort"
                          level="1"
                        >
                          Bubble Sort
                        </a>
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/quick-sort"
                          level="1"
                        >
                          Quick Sort
                        </a>
                      </ul>
                    </div>
                  </li>
                  <li
                    class="ps-menuitem-root ps-submenu-root css-1642kfo"
                  >
                    <a
                      class="ps-menu-button css-1bdadyq"
                      data-testid="ps-menu-button-test-id"
                    >
                      <span
                        class="ps-menu-label css-12w9als"
                      >
                        Binary Tree
                      </span>
                      <span
                        class="ps-submenu-expand-icon css-1cuxlhl"
                      >
                        <span
                          class="css-honxw6"
                        />
                      </span>
                    </a>
                    <div
                      class="ps-submenu-content css-z5rm24"
                      data-testid="ps-submenu-content-test-id"
                    >
                      <ul
                        class="css-ewdv3l"
                      >
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/binary-tree"
                          level="1"
                        >
                          BFS and DFS Traversal
                        </a>
                      </ul>
                    </div>
                  </li>
                  <li
                    class="ps-menuitem-root ps-submenu-root css-1642kfo"
                  >
                    <a
                      class="ps-menu-button css-1bdadyq"
                      data-testid="ps-menu-button-test-id"
                    >
                      <span
                        class="ps-menu-label css-12w9als"
                      >
                        Cryptography
                      </span>
                      <span
                        class="ps-submenu-expand-icon css-1cuxlhl"
                      >
                        <span
                          class="css-honxw6"
                        />
                      </span>
                    </a>
                    <div
                      class="ps-submenu-content css-z5rm24"
                      data-testid="ps-submenu-content-test-id"
                    >
                      <ul
                        class="css-ewdv3l"
                      >
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/caesar-cipher"
                          level="1"
                        >
                          Caesar Cipher
                        </a>
                      </ul>
                    </div>
                  </li>
                  <li
                    class="ps-menuitem-root ps-submenu-root css-1642kfo"
                  >
                    <a
                      class="ps-menu-button css-1bdadyq"
                      data-testid="ps-menu-button-test-id"
                    >
                      <span
                        class="ps-menu-label css-12w9als"
                      >
                        Computer Graphics
                      </span>
                      <span
                        class="ps-submenu-expand-icon css-1cuxlhl"
                      >
                        <span
                          class="css-honxw6"
                        />
                      </span>
                    </a>
                    <div
                      class="ps-submenu-content css-z5rm24"
                      data-testid="ps-submenu-content-test-id"
                    >
                      <ul
                        class="css-ewdv3l"
                      >
                        <a
                          class="d-block text-center p-1 text-decoration-none"
                          href="#/flood-fill"
                          level="1"
                        >
                          Flood Fill
                        </a>
                      </ul>
                    </div>
                  </li>
                </ul>
              </nav>
            </div>
          </aside>
        </div>
        <div
          class="col-md-9 col-lg-10"
        >
          <div
            class="waveWrapper waveAnimation"
          >
            <div
              class="waveWrapperInner bgTop"
            >
              <div
                class="wave waveTop"
              />
            </div>
            <div
              class="waveWrapperInner bgMiddle"
            >
              <div
                class="wave waveMiddle"
              />
            </div>
            <div
              class="waveWrapperInner bgBottom"
            >
              <div
                class="wave waveBottom"
              />
            </div>
          </div>
          <div
            class="wave-content"
          >
            <div
              class="row g-0 justify-content-center text-center p-3"
            >
              <div
                class="col-md-12"
              >
                <h3
                  class="text-white"
                >
                  Linear Search Algorithm
                </h3>
                <div
                  class="row justify-content-center"
                >
                  <div
                    class="col-md-6"
                  >
                    <div
                      class="mb-3 input-group"
                    >
                      <input
                        class="form-control"
                        disabled=""
                        value=""
                      />
                      <button
                        class="btn btn-primary"
                        type="button"
                      >
                        Random Fill
                      </button>
                    </div>
                    <div
                      class="mb-3 input-group"
                    >
                      <input
                        class="form-control"
                        min="1"
                        placeholder="Enter search criteria"
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0 p-3"
    >
      <div
        class="col-md-3 col-lg-2"
      >
        <aside
          class="ps-sidebar-root css-1qqt2yo"
          data-testid="ps-sidebar-root-test-id"
          style="z-index: 100;"
          width="230px"
        >
          <div
            class="ps-sidebar-container css-1o85esu"
            data-testid="ps-sidebar-container-test-id"
          >
            <h3
              class="text-white bg-primary text-center p-2"
            >
              Algorithms
            </h3>
            <nav
              class="ps-menu-root css-vj11vy"
            >
              <ul
                class="css-ewdv3l"
              >
                <li
                  class="ps-menuitem-root ps-submenu-root ps-open css-1642kfo"
                >
                  <a
                    class="ps-menu-button css-1bdadyq"
                    data-testid="ps-menu-button-test-id"
                  >
                    <span
                      class="ps-menu-label css-12w9als"
                    >
                      Searching
                    </span>
                    <span
                      class="ps-submenu-expand-icon css-1cuxlhl"
                    >
                      <span
                        class="css-jn69v9"
                        open=""
                      />
                    </span>
                  </a>
                  <div
                    class="ps-submenu-content css-edu7i6"
                    data-testid="ps-submenu-content-test-id"
                    open=""
                  >
                    <ul
                      class="css-ewdv3l"
                    >
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/linear-search"
                        level="1"
                      >
                        Linear Search
                      </a>
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/binary-search"
                        level="1"
                      >
                        Binary Search
                      </a>
                    </ul>
                  </div>
                </li>
                <li
                  class="ps-menuitem-root ps-submenu-root css-1642kfo"
                >
                  <a
                    class="ps-menu-button css-1bdadyq"
                    data-testid="ps-menu-button-test-id"
                  >
                    <span
                      class="ps-menu-label css-12w9als"
                    >
                      Sorting
                    </span>
                    <span
                      class="ps-submenu-expand-icon css-1cuxlhl"
                    >
                      <span
                        class="css-honxw6"
                      />
                    </span>
                  </a>
                  <div
                    class="ps-submenu-content css-z5rm24"
                    data-testid="ps-submenu-content-test-id"
                  >
                    <ul
                      class="css-ewdv3l"
                    >
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/bubble-sort"
                        level="1"
                      >
                        Bubble Sort
                      </a>
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/quick-sort"
                        level="1"
                      >
                        Quick Sort
                      </a>
                    </ul>
                  </div>
                </li>
                <li
                  class="ps-menuitem-root ps-submenu-root css-1642kfo"
                >
                  <a
                    class="ps-menu-button css-1bdadyq"
                    data-testid="ps-menu-button-test-id"
                  >
                    <span
                      class="ps-menu-label css-12w9als"
                    >
                      Binary Tree
                    </span>
                    <span
                      class="ps-submenu-expand-icon css-1cuxlhl"
                    >
                      <span
                        class="css-honxw6"
                      />
                    </span>
                  </a>
                  <div
                    class="ps-submenu-content css-z5rm24"
                    data-testid="ps-submenu-content-test-id"
                  >
                    <ul
                      class="css-ewdv3l"
                    >
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/binary-tree"
                        level="1"
                      >
                        BFS and DFS Traversal
                      </a>
                    </ul>
                  </div>
                </li>
                <li
                  class="ps-menuitem-root ps-submenu-root css-1642kfo"
                >
                  <a
                    class="ps-menu-button css-1bdadyq"
                    data-testid="ps-menu-button-test-id"
                  >
                    <span
                      class="ps-menu-label css-12w9als"
                    >
                      Cryptography
                    </span>
                    <span
                      class="ps-submenu-expand-icon css-1cuxlhl"
                    >
                      <span
                        class="css-honxw6"
                      />
                    </span>
                  </a>
                  <div
                    class="ps-submenu-content css-z5rm24"
                    data-testid="ps-submenu-content-test-id"
                  >
                    <ul
                      class="css-ewdv3l"
                    >
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/caesar-cipher"
                        level="1"
                      >
                        Caesar Cipher
                      </a>
                    </ul>
                  </div>
                </li>
                <li
                  class="ps-menuitem-root ps-submenu-root css-1642kfo"
                >
                  <a
                    class="ps-menu-button css-1bdadyq"
                    data-testid="ps-menu-button-test-id"
                  >
                    <span
                      class="ps-menu-label css-12w9als"
                    >
                      Computer Graphics
                    </span>
                    <span
                      class="ps-submenu-expand-icon css-1cuxlhl"
                    >
                      <span
                        class="css-honxw6"
                      />
                    </span>
                  </a>
                  <div
                    class="ps-submenu-content css-z5rm24"
                    data-testid="ps-submenu-content-test-id"
                  >
                    <ul
                      class="css-ewdv3l"
                    >
                      <a
                        class="d-block text-center p-1 text-decoration-none"
                        href="#/flood-fill"
                        level="1"
                      >
                        Flood Fill
                      </a>
                    </ul>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
        </aside>
      </div>
      <div
        class="col-md-9 col-lg-10"
      >
        <div
          class="waveWrapper waveAnimation"
        >
          <div
            class="waveWrapperInner bgTop"
          >
            <div
              class="wave waveTop"
            />
          </div>
          <div
            class="waveWrapperInner bgMiddle"
          >
            <div
              class="wave waveMiddle"
            />
          </div>
          <div
            class="waveWrapperInner bgBottom"
          >
            <div
              class="wave waveBottom"
            />
          </div>
        </div>
        <div
          class="wave-content"
        >
          <div
            class="row g-0 justify-content-center text-center p-3"
          >
            <div
              class="col-md-12"
            >
              <h3
                class="text-white"
              >
                Linear Search Algorithm
              </h3>
              <div
                class="row justify-content-center"
              >
                <div
                  class="col-md-6"
                >
                  <div
                    class="mb-3 input-group"
                  >
                    <input
                      class="form-control"
                      disabled=""
                      value=""
                    />
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Random Fill
                    </button>
                  </div>
                  <div
                    class="mb-3 input-group"
                  >
                    <input
                      class="form-control"
                      min="1"
                      placeholder="Enter search criteria"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
